src/super_context.ml

```diff
@@ -631,10 +631,23 @@ module Action = struct
     let expand var syntax_version =
       let loc = String_with_vars.Var.loc var in
       let key = String_with_vars.Var.full_name var in
+      let path_with_dep s =
+        Some (path_exp (Path.relative dir s) )
+      in
       match String_with_vars.Var.destruct var with
-      | Pair ("exe"     , s) -> Some (path_exp (map_exe (Path.relative dir s)))
-      | Pair ("path"    , s) -> Some (path_exp          (Path.relative dir s) )
-      | Pair ("bin"     , s) -> begin
+      | Pair ("exe", s) -> Some (path_exp (map_exe (Path.relative dir s)))
+      | Pair ("path", s) when syntax_version < (1, 0) ->
+        path_with_dep s
+      | Pair ("dep", s) when syntax_version >= (1, 0) ->
+        path_with_dep s
+      | Pair ("dep", s) ->
+          Loc.fail
+            loc
+            "${dep:%s} is not supported in jbuild files.\n\
+             Did you mean: ${path:%s}"
```

Comment:
In general dune prints: `Hint: Did you mean...`

Reply:
:+1: added this. It looks a bit weird because the question mark is missing, but at the same time if we add a question mark at the end it looks like it's part of the suggestion.

Reply:
I see. What about this:

```
Hint: Did you mean ${path:xxx} instead?
```

Reply:
Looks good, I just made this change.